src/drivers/intel/hda/hda-dma.c

```diff
@@ -829,12 +829,16 @@ static int hda_dma_link_check_xrun(struct dma_chan_data *chan)
 {
 	uint32_t dgcs = dma_chan_reg_read(chan, DGCS);
 
-	if (chan->direction == DMA_DIR_MEM_TO_DEV && dgcs & DGCS_BUR) {
+	if (chan->direction == DMA_DIR_MEM_TO_DEV && (dgcs & DGCS_BUR)) {
 		tr_err(&hdma_tr, "hda_dma_link_check_xrun(): underrun detected");
 		dma_chan_reg_update_bits(chan, DGCS, DGCS_BUR, DGCS_BUR);
-	} else if (chan->direction == DMA_DIR_DEV_TO_MEM && dgcs & DGCS_BOR) {
+		return -ENODATA;
+	}
+
+	if (chan->direction == DMA_DIR_DEV_TO_MEM && (dgcs & DGCS_BOR)) {
 		tr_err(&hdma_tr, "hda_dma_link_check_xrun(): overrun detected");
 		dma_chan_reg_update_bits(chan, DGCS, DGCS_BOR, DGCS_BOR);
+		return -ENODATA;
```

Comment:
doesn't this need additional changes in the caller as mentioned by @lyakh ?

Reply:
I think his comment was for my host changes and I fixed that in my other PR. This one is correct isnt it?

Reply:
I can't recall precisely the issue... Just wanted to check that this change is legit.